RA-09-CallBack-for-Categories

DenisGubernatorov opened the pull request:
1) Добавлен лэйаут для экрана со списком рецептов в категории
2) Реализация заготовки класса фрагмента для списка рецептов
3) Добавление слушателя для вью категорий.
4) Реализация перехода на экран со списком рецептов категории.

DenisGubernatorov committed:
RA-09-CallBack-For-Categories
1) Добавлен лэйаут для экрана со списком рецептов в категории
2) Реализация заготовки класса фрагмента для списка рецептов

DenisGubernatorov committed:
RA-09-CallBack-for-Categories

1) Добавление слушателя для вью категорий.
2) Реализация перехода на экран со списком рецептов категории.

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -31,9 +31,23 @@ class CategoriesListFragment : Fragment() {
+        val recipesListFragment = RecipesListFragment()
+        parentFragmentManager.beginTransaction()
+            .replace(R.id.mainContainer, recipesListFragment).addToBackStack(null).commit()
+
```
Сейчас ты используешь старый джавовский способ перехода между фрагментами. В прошлых уроках мы уже делали через commit{} и `replace<...>`.
Также нужно заменить replace на add, сейчас при нажатии на кнопку назад - приложение закроется, правильнее будет, чтоб пользователь мог вернуться назад

DenisGubernatorov replied:
В моем подходе кнопка "Назад" тоже работала. Проверял. )) И, если честно, то не очень понятно почему добавление, а не замена. По сути мы один фрагмент заменяем другим. Визуально RecipesListFragment перекроет CategoriesListFragment. А addToBackStack добвляет стек возврата. Таким образом можно вернуться к предуыдущему фрагменту. А add же добавляет фрагмент к другим,если они есть, в текущий контейнер.  Если я что-то не совсем правильно описал, поправь пожалуйста.

aGroupoid replied:
Сейчас на простом сценарии трудно увидеть разницу между add и replace. При replace твой фрагмент полностью умирает, вызывается onDestroyView. Когда нажимаешь назад, он создается заново и отрисовывает его по новой. В add он сохраняет предыдущий экран. В прошлом комменте немного не так описал поведение.
Можешь поиграться сделать рутовые вьюхи на 50% прозрачными, чтобы видеть что лежит под ними. Либо же написать простой кликер с переходом через add и replace. С add количество кликов сохранится при нажатии назад, с replace - будет равно 0

DenisGubernatorov committed:
RA-09-CallBack-for-Categories

1) Доработки согласно рекомендациям в ПР https://github.com/DenisGubernatorov/RecipesApp/pull/7#discussion_r1867961651

aGroupoid reviewed (approved):
Молодец, всё верно!
